Fang-4-Group/Graduation-Project
# Yunsi/gp 117/prosecc interaction

## Changes
- Add sql to create interaction tables
- Add db crud function
  - add interaction
  - add interaction detail
  - update field ( to change Viewed, Grouped, Selected value)
  - get interaction
- Add four api for above function
  - add interaction (including interaction and interaction detail)
  - update_viewed
  - update_grouped
  - update_selected
  - get_interaction
- API testing function

## Test Step
1. run `docker-compose up -d`
2. go to postman, click add_interaction, you will receive
```
{
    "role": "Young",
    "message": "Initial detail for Interaction ID 1 inserted successfully"
}
```
3. then click update_viewed, you will receive
```
{
    "message": "Update Successfully"
}
```

you can also change the path to update_grouped or update_selected, and see the corresponding output of get_interaction.

4.  finally click get_interaction, you will receive
```
[
    {
        "Detail_ID_y": 2,
        "Interaction_ID_y": 1,
        "House_ID": 8,
        "Viewed": 1,
        "Grouped": 0,
        "Selected": 0
    }
]
```
